tests/test_types.py

```diff
@@ -59,7 +59,47 @@ def test_struct_deduplication():
     dhdr = angr.types.define_struct('struct dhdr { struct ahdr a; struct bhdr b; struct chdr c;}')
     assert dhdr.fields['a'].fields
 
+def test_parse_type():
+    int_ptr = angr.types.parse_type('int *')
```

Comment:
can there be a test that runs parse_type with a name involved, like `int *x`?

Reply:
the other major thing missing representation in this test is function pointers

Reply:
I have added tests with function pointer type expression and variable declaration.